shared/actions/kbfs/index.platform.desktop.js

```diff
@@ -55,24 +46,71 @@ function openInDefault (openPath: string): Promise<*> {
           reject(err)
           return
         }
+        console.log('Opened directory:', openPath)
         resolve()
       })
     })
   })
 }
 
+function isDirectory (openPath: string): Promise<boolean> {
+  return new Promise((resolve, reject) => {
+    fs.stat(openPath, (err, stats) => {
+      if (err) {
+        reject(new Error(`Unable to open/stat file: ${openPath}`))
+        return
+      }
+      if (stats.isFile()) {
+        resolve(false)
+      } else if (stats.isDirectory()) {
+        resolve(true)
+      } else {
+        reject(new Error(`Unable to open: Not a file or directory`))
+      }
+    })
+  })
+}
+
+function openInDefault (openPath: string): Promise<*> {
+  return new Promise((resolve, reject) => {
+    console.log('openInDefault:', openPath)
+    // Path resolve removes any ..
+    openPath = path.resolve(openPath)
+    // Paths MUST start with defaultKBFSPath
+    if (!openPath.startsWith(Constants.defaultKBFSPath)) {
+      reject(new Error(`openInDefault requires ${Constants.defaultKBFSPath} prefix: ${openPath}`))
+      return
+    }
+
+    isDirectory(openPath).then((isDir) => {
+      if (isDir) {
+        openInDefaultDirectory(openPath).then(resolve).catch(reject)
```

Comment:
:sheep: `.then(resolve, reject)` is better because the `.catch` might cover something in the resolve call.